lib/handler/range.c

```diff
@@ -0,0 +1,112 @@
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "h2o.h"
+
+
+#define CHECK_EOF()                             \
+    if (buf == buf_end) {                       \
+        *ret = -2;                              \
+        return NULL;                            \
+    }
+
+#define EXPECT_CHAR(ch)                         \
+    CHECK_EOF();                                \
+    if (*buf++ != ch) {                         \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+#define CHECK_OVERFLOW(range)                   \
+    if (range == SIZE_MAX) {                    \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+size_t *process_range(h2o_mem_pool_t *pool, h2o_iovec_t *range_value, size_t file_size, size_t* ret)
```

Comment:
RFC 7233 defines that 
```
byte-range-set = *( "," OWS ) ( byte-range-spec /
suffix-byte-range-spec ) *( OWS "," [ OWS ( byte-range-spec /
suffix-byte-range-spec ) ] )
```
So I think I should
* check `,[ \t]*` at the top of the do-while block (and skip). If `[ \t]*` is skipped when last single range parsing ended and no `,` found, error occurred.
* skip `[ \t]*` at each exit from single range parsing